vision/app.py

```diff
@@ -80,91 +82,107 @@ def run_image(self):
-				except:
-					lowerThreshold = self.lower
-					upperThreshold = self.upper
-				print upperThreshold, lowerThreshold
-			else:
-				im = cameraRear.read()
-				try:
-					lowerThreshold = np.array([nt_utils.get_number("rear_lower_blue"), nt_utils.get_number("rear_lower_green"), nt_utils.get_number("rear_lower_red")])
-					upperThreshold = np.array([nt_utils.get_number("rear_upper_blue"), nt_utils.get_number("rear_upper_green"), nt_utils.get_number("rear_upper_red")])
-				except:
-					lowerThreshold = self.lower
-					upperThreshold = self.upper
-
-			if im is not None:
-				im = cv2.resize(im, (600, 480), 0, 0)
-				try:
-					blob, im_mask = cv_utils.get_blob(im, lowerThreshold, upperThreshold)
-				except TypeError:
-					blob, im_mask = cv_utils.get_blob(im, self.lower, self.upper)
-				if blob is not None:
-					x1, y1, w1, h1 = cv2.boundingRect(blob[0])
-					x2, y2, w2, h2 = cv2.boundingRect(blob[1])
-
-					area1 = w1 * h1
-					area2 = w2 * h2
-					totalArea = area1 + area2
-					if (totalArea > self.min_area) and (totalArea < self.max_area):
-						if verbose:
-							print("[Blob] x: %d, y: %d, width: %d, height: %d, total area: %d" % (x1, y1, w1, h1, totalArea))
-
-						offset_x, offset_y = cv_utils.process_image(im, x1, y1, w1, h1, x2, y2, w2, h2)
-
-						nt_utils.put_number("offset_x", offset_x)
-						nt_utils.put_number("offset_y", offset_y)
-						nt_utils.put_boolean("blob_found", True)
-						nt_utils.put_number("blob1_size", w1 * h1)
-						nt_utils.put_number("blob2_size", w2 * h2)
-					else:
-						nt_utils.put_boolean("blob_found", False)
-
-					if self.display:
-						# Draw image details
-						im = cv_utils.draw_images(im, x1, y1, w1, h1, True)
-						im = cv_utils.draw_images(im, x2, y2, w2, h2, False)
-
-						# Show the images
-						cv2.imshow("Original", im)
-						cv2.imshow("Mask", im_mask)
+			cv2.destroyAllWindows()
+
+	def get_frame(self):
+		if nt_utils.get_boolean("shutdown"):
+			os.system("shutdown -H now")
+			return
+
+		if nt_utils.get_boolean("front_camera"):
+			im = self.cameraFront.read()
+			try:
+				lowerThreshold = np.array([nt_utils.get_number("front_lower_blue"), nt_utils.get_number("front_lower_green"), nt_utils.get_number("front_lower_red")])
+				upperThreshold = np.array([nt_utils.get_number("front_upper_blue"), nt_utils.get_number("front_upper_green"), nt_utils.get_number("front_upper_red")])
+			except:
+				lowerThreshold = self.lower
+				upperThreshold = self.upper
+
+		else:
+			im = self.cameraRear.read()
+			try:
+				lowerThreshold = np.array([nt_utils.get_number("rear_lower_blue"), nt_utils.get_number("rear_lower_green"), nt_utils.get_number("rear_lower_red")])
+				upperThreshold = np.array([nt_utils.get_number("rear_upper_blue"), nt_utils.get_number("rear_upper_green"), nt_utils.get_number("rear_upper_red")])
+			except:
+				lowerThreshold = self.lower
+				upperThreshold = self.upper
+
+		if im is not None:
+			im = cv2.resize(im, (600, 480), 0, 0)
```

Comment:
I'm sure there's a good reason for this number to be 600, but most camera's aspect ratio would be 4:3 which would make the size 640 x 480.

Reply:
That's a good point. I think we might have used this last year accidentally for some reason.

Reply:
Also, use constants instead of these magic numbers.